dashboard/test/models/activity_test.rb

```diff
@@ -1,7 +1,65 @@
 require 'test_helper'
+require 'securerandom'
+require 'sqs/synchronous_queue'
 
 class ActivityTest < ActiveSupport::TestCase
-  # test "the truth" do
-  #   assert true
-  # end
+
+  # A test handler which allows deferred handling of asynchronous messages.
+  # It records the messages it is alled with an then calls the underlying
+  # AsyncActivityHandler when `run_queued_messages` is called.
+  class DeferredActivityHandler
+    def initialize
+      @messages = []
+      @activity_handler = Activity::AsyncHandler.new
+    end
+
+    def handle(messages)
+      @messages += messages
+    end
+
+    def execute_async_writes
+      @activity_handler.handle(@messages)
+      @messages.clear
+    end
+  end
+
+  def test_create_async!
+    handler = DeferredActivityHandler.new
+    queue = SQS::SynchronousQueue.new(handler)
+
+    student = create :student
+    level = create :applab
+    level_source = LevelSource.find_identical_or_create(level, SecureRandom.uuid)
+    activity = Activity.create_async!(queue,
+                                      {level: level,
+                                       user: student,
+                                       level_source: level_source})
+    # The new activity returned by create_async doesn't have an id or timestamps yet, but it does
+    # have other attributes.
+    assert_nil activity.id
+    assert_nil activity.created_at
+    assert_nil activity.updated_at
+    assert_equal student.id, activity.user_id
+    assert_equal level.id, activity.level_id
+    assert_equal level_source.id, activity.level_source_id
+
+    # We shouldn't be able to find the activity in the database
+    # because the async write hasn't happened yet.
+    activity_finder = Activity.where(user_id: student.id)
+                          .where(level_id: level.id)
```

Comment:
Style/DotPosition: Place the . on the previous line, together with the method call receiver.